src/data/starredSlice.js

```diff
@@ -0,0 +1,22 @@
+import { createSlice } from "@reduxjs/toolkit"
+
+const starredSlice = createSlice({
+    name: 'starred',
+    initialState: {
+        starredMovies: []
+    },
+    reducers: {
+        starMovie: (state, action) => {
+            state.starredMovies = [action.payload, ...state.starredMovies]
+        },
+        unstarMovie: (state, action) => {
+            const indexOfId = state.starredMovies.findIndex(key => key.id === action.payload.id)
```

Comment:
I noticed a potential enhancement that might improve the performance of removing items from the starredMovies array. Currently, the code employs the findIndex method followed by a splice operation to remove the selected item. Would you mind if I proposed an alternative approach using the filter method?

```
state.starredMovies = state.starredMovies.filter(movie => movie.id !== action.payload.id);
```

It could potentially offer improved performance.